### PR TITLE
Add unit tests for replay location engine

### PR DESCRIPTION
**Pull request by kmadsen**
### Description

Adding unit tests where there are none

### Screenshots or Gifs

Before
Screen Shot 2020-11-12 at 12 10 47 PM

[After](https://codecov.io/gh/mapbox/mapbox-navigation-android/tree/km-replay-location-engine-tests/libnavigation-core/src/main/java/com/mapbox/navigation/core/replay)
Screen Shot 2020-11-12 at 1 57 54 PM

### CODE REVIEW

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/replay/ReplayLocationEngineTest.kt [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        assertEquals(-121.452869, lastLocation.longitude, 0.000001)
+        assertEquals("ReplayLocationEngineTest", lastLocation.provider)
+        assertEquals(158.32, lastLocation.altitude, 0.01)
+        assertEquals(12.3f, lastLocation.accuracy, 0.01f)
+        assertEquals(287.4f, lastLocation.bearing, 0.01f)
+        assertEquals(12.2f, lastLocation.speed, 0.01f)
+    }
+
+    @Test
+    fun `should map monotonic time`() {
```
Comment (kmadsen) [inline libnavigation-core/src/test/java/com/mapbox/navigation/core/replay/ReplayLocationEngineTest.kt:183]: left out the System.currentTimeMillis() test because https://github.com/mockk/mockk/issues/98

Worth considering `Calendar.getInstance().timeInMillis` instead of `Date()`
But it becomes more complicated with timezones

My preferred approach for testing time, is to create a `MapboxTime` interface that can be mocked. there's a couple open source options but that would mainly be motivated by incorporating NTP time
- https://github.com/instacart/truetime-android
- https://github.com/lyft/Kronos-Android 

conclusion: leaving that test out because it's a bigger project with less reward right now

Comment (Guardiola31337) [reply libnavigation-core/src/test/java/com/mapbox/navigation/core/replay/ReplayLocationEngineTest.kt:183]: > My preferred approach for testing time, is to create a `MapboxTime` interface that can be mocked.

Same here.

**Review by Guardiola31337 — APPROVED** [review]
Amazing test coverage ❤️
